### PR TITLE
Issue 178 service account authentication docs

### PR DESCRIPTION
**Pull request by sbryzak**
Documented service account authentication
Fixes https://github.com/fabric8-services/fabric8-auth/issues/178 and https://github.com/fabric8-services/fabric8-auth/issues/176

### CODE REVIEW

**Comment by fabric8cd** [comment]
@sbryzak snapshot fabric8-auth image is available for testing.  `docker pull docker.io/fabric8/fabric8-auth:SNAPSHOT-PR-187-1`

File: docs/source/reference.adoc [hunk 1]
```diff
@@ -1,3 +1,126 @@
...
+:toc:
 
-This is the Reference documentation for fabric8-auth.
+This is the Reference documentation for fabric8-auth. You may find it useful if you wish to deploy and configure the service, or to interact
+with its authentication and authorization services via its REST endpoints.  If you wish to understand more about the inner workings of fabric8-auth and be
+able to build and/or develop the project, please refer to the link:developer.html[Developer documentation] instead.
+
+== Service Accounts
+
+Service accounts are used to authenticate other services in order for them to interact with the fabric8-auth permission (and other) endpoints.
```
Comment (alexeykazakov) [inline docs/source/reference.adoc]: It's not for interacting with fabric8-auth endpoints only. It's for service-to-service communication in general.

**Comment by fabric8cd** [comment]
@sbryzak snapshot fabric8-auth image is available for testing.  `docker pull docker.io/fabric8/fabric8-auth:SNAPSHOT-PR-187-2`

**Comment by joshuawilson** [comment]
fixes #178

**Comment by fabric8cd** [comment]
@sbryzak snapshot fabric8-auth image is available for testing.  `docker pull docker.io/fabric8/fabric8-auth:SNAPSHOT-PR-187-3`

**Pull request merged**